filters.py

```diff
@@ -0,0 +1,154 @@
+        except Exception:
+            await message.edit("<b>Filter ''{}'' not found in this chat</b>".format(filtern))
+
+    async def stopallcmd(self, message):
+        """Clears out the filter list."""
+        filters = self._db.get("FilterModule", "filters")
+        chatid = str(message.chat_id)
+        try:
+            del filters[chatid]
+            self._db.set("FilterModule", "filters", filters)
+            await message.edit("<b>All filters successfully removed from the chat.</b>")
+        except Exception:
+            await message.edit("<b>There are no filters to clear out in this chat.</b>")
+
+    async def filterscmd(self, message):
+        """Shows saved filters."""
+        filters = ""
+        filt = self._db.get("FilterModule", "filters")
+        chatid = str(message.chat_id)
+        try:
+            for i in filt[chatid]:
+                filters += "<b> -  " + str(i) + "</b>\n"
+                pass
+        except Exception:
+            pass
+        filterl = "<b>Text that you filtered in this chat: </b>\n\n" + filters
+        if filters:
+            await message.edit(filterl)
+        else:
+            await message.edit("<b>No filters found in this chat.</b>")
+
+    async def watcher(self, message):
+        # This is to execute cmds
+        exec = True
+        for load in self.allloaders:
+            if load.client is self._client:
+                break
+        args = str(message.text.lower())
+        filters = self._db.get("FilterModule", "filters")
+        chatid = str(message.chat_id)
+        try:
+            current_prefix = self._db.get("friendly-telegram.main", "command_prefix")
+        except Exception:
+            current_prefix = "."
+        if filters and chatid in filters:
+            for key in filters[chatid]:
+                if key.lower() in args.lower():
+                    id = filters[chatid][key]
+                    value = await self._db.fetch_asset(int(id))
+                    if not value.media and not value.web_preview:
+                        if value.text.startswith(current_prefix):
+                            cmd = value.message[1::]
+                        if value.text.startswith(current_prefix * 2):
+                            cmd = value.message[2::]
+                        if not value.text.startswith(current_prefix):
+                            exec = False
+                            cmd = value.message
+                        if not exec:
+                            await message.client.send_message(int(chatid), value.message, reply_to=message.id)
+                            return
+                        else:
+                            cmdspl = cmd.lower().split(" ")
+                            respond = await message.client.send_message(int(chatid), cmd, reply_to=message.id)
+                            try:
+                                await load.dispatch(cmdspl[0], respond)
```

Comment:
WHAT THE FUCK ARE YOU CRAZY!?!?!?!?!?!!?!?!?!?!?! DONT SHIP THIS MODULE ANYWHERE!!!!!!!!!!!!!!!!!!!!!!!!!!!!